File: Daily Challanges/2026-06-June/2026-06-14.py
def is_valid_card(card_number):
    total = 0
    for i, char in enumerate(reversed(card_number)):
        digit = int(char)
        if i % 2 == 1:
            digit *= 2
            if digit > 9:
                digit -= 9
        total += digit
    return total % 10 == 0


print(is_valid_card("4532015112830366"))
print(is_valid_card("5425233430109903"))
print(is_valid_card("371449635398431"))
print(is_valid_card("6011111111111117"))
print(is_valid_card("4532015112830367"))
print(is_valid_card("1234567890123456"))
print(is_valid_card("4532015112830368"))

File: Daily Challanges/2026-06-June/2026-06-14.js
function isValidCard(cardNumber) {
  let total = 0;
  for (let i = 0; i < cardNumber.length; i++) {
    let digit = parseInt(cardNumber[cardNumber.length - 1 - i], 10);
    if (i % 2 === 1) {
      digit *= 2;
      if (digit > 9) {
        digit -= 9;
      }
    }
    total += digit;
  }
  return total % 10 === 0;
}

console.log(isValidCard("4532015112830366"));
console.log(isValidCard("5425233430109903"));
console.log(isValidCard("371449635398431"));
console.log(isValidCard("6011111111111117"));
console.log(isValidCard("4532015112830367"));
console.log(isValidCard("1234567890123456"));
console.log(isValidCard("4532015112830368"));
